user-interface/0001-WALL-wallet.md

```diff
@@ -0,0 +1,179 @@
+
+- **must** be prompted to download a newer version, and given a link to get latest on github <a name="0001-WALL-014" href="#0001-WALL-014">0001-WALL-014</a>
+- **must** be warned if the version I am using is not compatible with the version of Vega on the selected network, and I am given a link to get latest on github <a name="0001-WALL-015" href="#0001-WALL-015">0001-WALL-015</a>
+
+... so the version of the wallet app I am using works with the network I am using
+
+## Log in to a wallet
+When using a given wallet, I...
+
+- **must** enter wallet name + passphrase only once per "session" <a name="0001-WALL-016" href="#0001-WALL-016">0001-WALL-016</a>
+
+... so that other users of my machine can not use my wallet, and I am not asked to re-enter frequently
+
+## Connecting to Dapps
+When a dapp requests use of a wallet, I...
+
+- **must** be prompted to either select a wallet or dismiss the prompt  <a name="0001-WALL-017" href="#0001-WALL-017">0001-WALL-017</a>
+- `TODO` am required to select what keys of a wallet to grant access too <!--<a name="0001-WALL-018" href="#0001-WALL-018">0001-WALL-018</a>-->
+  - `TODO:` **must** be able to select whole wallet (so that new keys are automatically shared)  <!--<a name="0001-WALL-019" href="#0001-WALL-019">0001-WALL-019</a>-->
+  - `TODO:` **must** select specific [keys](7001-DATA-data_display.md#public-keys) <!-- <a name="0001-WALL-020" href="#0001-WALL-020">0001-WALL-020</a> -->
+  - `TODO` tainted keys **must** be shown as tainted <a name="0001-WALL-021" href="#0001-WALL-021">0001-WALL-021</a>
+- **must** enter wallet passphrase before wallet details are shared <a name="0001-WALL-022" href="#0001-WALL-022">0001-WALL-022</a>
+- **must** be able to retrospectively revoke Dapp's access to Wallet/keys <a name="0001-WALL-023" href="#0001-WALL-023">0001-WALL-023</a>
+
+... so that I can control what public keys are shared with a dapp and what dapps can prompt me to sign transactions 
+
+## Approving transactions
+When a dapp sends a transaction to the wallet for signing and broadcast, I...
+
+- **must** be prompted to approve, reject or ignore the transaction (if auto approve is not on) <a name="0001-WALL-024" href="#0001-WALL-024">0001-WALL-024</a>
+- **must** see the details of the transaction. See [details of transaction](#transaction-detail). <a name="0001-WALL-025" href="#0001-WALL-025">0001-WALL-025</a>
+- **must** see any ignored/dismissed (not rejected or approved) in a transactions area (e.g. history). See [Transactions](#transactions).
+
+... so I can verify that the transaction being sent is the one I want
+
+## View keys balances + positions/accounts
+When looking for a specific balance or asset on a given wallet and network, I...
+
+- **must** see key total balances of an asset for each key <a name="0001-WALL-027" href="#0001-WALL-027">0001-WALL-027</a>
+- **must** see a breakdown of all accounts for an asset on a key <a name="0001-WALL-028" href="#0001-WALL-028">0001-WALL-028</a>
```

Comment:
@barnabee the WIP to show assets on a key in the wallet app will only request assets for the selected key.
Perhaps this is sufficient to reduce leaking of info until we've built some configuration that allows you to control what keys request balances?

Another note: the API doesn't give a total balance, only each of the accounts for a party. So the UI will need to sum them up on the client. We could make it so that it doesn't stream these but fetches at a particular rate.

Reply:
changed to should in https://github.com/vegaprotocol/specs/pull/1181/commits/a86487963de0dd2b33df926308a1987e0bc661f1

Reply:
https://github.com/vegaprotocol/specs/commit/9ab47aeab2b167ee147e7de4c27a4150fa7f3f54 with 
> - **must** be able to control whether the wallet app queries for data (e.g. asset balances) on each key (to prevent info leaking about what keys belong to a wallet) <a name="0001-WALL-078" href="#0001-WALL-078">0001-WALL-078</a>

Should resolve this

Reply:
I think the minimum in the GUI app is a warning about fetching if no control, so if you wish you could update this to must warn and should allow control. The CLI app must not do any outbound connections like this so can be the option for the privacy conscious.

In any case the above resolves it so I am marking as resolved also.